src/substrait/textplan/SymbolTablePrinter.cpp

```diff
@@ -5,17 +5,244 @@
+        return "opt_i32";
+      return "i32";
+    case ::substrait::proto::Type::kI64:
+      if (type.i64().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE)
+        return "opt_i64";
+      return "i64";
+    case ::substrait::proto::Type::kFp32:
+      if (type.fp32().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE)
+        return "opt_fp32";
+      return "fp32";
+    case ::substrait::proto::Type::kFp64:
+      if (type.fp64().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE)
+        return "opt_fp64";
+      return "fp64";
+    case ::substrait::proto::Type::kString:
+      if (type.string().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE)
+        return "opt_string";
+      return "string";
+    case ::substrait::proto::Type::kDecimal:
+      if (type.string().nullability() ==
+          ::substrait::proto::Type::NULLABILITY_NULLABLE)
+        return "opt_decimal";
+      return "decimal";
+    case ::substrait::proto::Type::kVarchar:
+      return "varchar";
+    case ::substrait::proto::Type::kFixedChar:
+      return "fixedchar";
+    case ::substrait::proto::Type::kDate:
+      return "date";
+    case ::substrait::proto::Type::KIND_NOT_SET:
+    default:
+      return "UNSUPPORTED_TYPE";
+  }
+};
+
+std::string relationToText(
+    const SymbolTable& symbolTable,
+    const SymbolInfo& info) {
+  auto relation = ANY_CAST(const ::substrait::proto::Rel*, info.blob);
+
+  PlanPrinterVisitor printer(symbolTable);
+  return printer.printRelation(info.name, relation);
+}
+
+std::string outputPipelinesSection(const SymbolTable& symbolTable) {
+  // TODO: Implement.
+  return "";
+}
+
+std::string outputRelationsSection(const SymbolTable& symbolTable) {
+  std::stringstream text;
+  bool hasPreviousText = false;
+  for (const SymbolInfo& info : symbolTable) {
+    if (info.type != SymbolType::kRelation)
+      continue;
+    // TODO: Put handling this into the PlanPrinterVisitor.
+    if (hasPreviousText)
+      text << "\n";
+    text << relationToText(symbolTable, info);
+    hasPreviousText = true;
   }
+  return text.str();
+}
 
+std::string outputSchemaSection(const SymbolTable& symbolTable) {
+  std::stringstream text;
+  bool hasPreviousText = false;
+  for (const SymbolInfo& info : symbolTable) {
+    if (info.type != SymbolType::kSchema)
+      continue;
+
+    if (hasPreviousText)
+      text << "\n";
+
+    const auto* schema =
+        ANY_CAST(const ::substrait::proto::NamedStruct*, info.blob);
+    text << "schema " << info.name << " {\n";
+    int name_idx = 0;
+    int types_idx = 0;
+    while (name_idx < schema->names_size() &&
+           types_idx < schema->struct_().types_size()) {
+      text << "  " << schema->names(name_idx);
+      text << " " << typeToText(schema->struct_().types(types_idx));
+      text << ";\n";
+      ++name_idx;
+      ++types_idx;
+    }
+    text << "}\n";
```

Comment:
This is where the fun begins.  I am open to switching to use std::endl throughout.  It will affect the tests as different platforms will now be emitting different line endings.  Windows is generally okay with the lack of a carriage return but it often depends on the editor.  I believe it was pre-Unix Macintoshes that used only a CR as an end of line marker would have significant trouble here.  The other consideration is how shareable do we want the resulting text output.